### PR TITLE
Support using `wslview` as a browser to open web url

### PR DESCRIPTION
**Pull request by G-Rath**
[`wslview`](http://manpages.ubuntu.com/manpages/eoan/man1/wslview.1.html) is a utility that ships with the newer versions of WSL as a way of opening links in the default web browser on the Windows host.

### CODE REVIEW

**Comment by Kriechi** [comment]
👍  from me. @mhils ?

**Comment by mhils** [comment]
Yes, this look great. @G-Rath, you tested this locally I presume? I can't reproduce this, but I'm on a rather older WSL1 installation.

**Comment by G-Rath** [comment]
Yeah I used this locally and it worked fine.

You can grab `wslview` from [here](https://github.com/wslutilities/wslu) - that repo is installed now as part of standard wsl from what I can tell.

**Comment by mhils** [comment]
Thanks, very useful. I'm still curious how you get it to work. Here's the output from my system:
```
λ python3 --version
Python 3.8.0

λ which wslview
/usr/bin/wslview

λ python3 -c "import webbrowser; webbrowser.get('wslview')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.8/webbrowser.py", line 65, in get
    raise Error("could not locate runnable browser")
webbrowser.Error: could not locate runnable browser
```

FWIW, `webbrowser.get('wslview %s')` does work for me, how does plain `wslview` as a browser work for you?

**Comment by G-Rath** [comment]
> FWIW, webbrowser.get('wslview %s') does work for me

That would be because `wslview` expects a URL :)

i.e 

```
wslview https://www.google.com
```

**Comment by mhils** [comment]
Well yes. Let me rephrase: With your PR applied, mitmproxy will never pick up wslview on my system because `webbrowser.get('wslview')` errors.

- If this does work for you, what are you doing differently? Why is your webbrowser.get call succeeding
- If it does not work for you, please add the " %s" part the the browser alternatives list so that it's properly picked up. :)

**Comment by G-Rath** [comment]
Ah ok I'm following you now.

Does calling `wslview https://www.google.com` in the commandline work for you at all? If not, I'd say you've not got a web browser installed that `wslview` can find.

What does your PATH look like?

**Comment by mhils** [comment]
`wslview https://www.google.com` works as expected. It's just Python/mitmproxy which doesn't find wslview, because `webbrowser.get('wslview')` doesn't return a browser object a few lines below your PR addition.

**Comment by mhils** [comment]
What's your output of `python3 -c "import webbrowser; webbrowser.get('wslview')"`?

**Comment by G-Rath** [comment]
My output is nothing - it exists with a successful error code (0).

**Comment by G-Rath** [comment]
what's the output if you run just `wslview` by itself?

**Comment by mhils** [comment]
All as expected:
```
λ wslview
[error]No input, aborting
λ wslview https://example.com/
λ echo $status
0
```

Which Python version are you on? Are you somehow monkeypatching wslview into your webbrowser browsers? I'm taking the following code path on my system:

https://github.com/python/cpython/blob/a3d6d23a923d62a713b9e4536d5f7ce7313d99c7/Lib/webbrowser.py#L37
https://github.com/python/cpython/blob/a3d6d23a923d62a713b9e4536d5f7ce7313d99c7/Lib/webbrowser.py#L60
https://github.com/python/cpython/blob/a3d6d23a923d62a713b9e4536d5f7ce7313d99c7/Lib/webbrowser.py#L105
https://github.com/python/cpython/blob/a3d6d23a923d62a713b9e4536d5f7ce7313d99c7/Lib/webbrowser.py#L124

**Comment by G-Rath** [comment]
Python is not my primary language, so I wouldn't even know where to start on monkey patches :) 

The only monkey patch I have done is the change in this PR, to the `mitmproxy` package.

I'm running Python 3.8.1 (default, Mar 10 2020, 15:49:49) [GCC 7.4.0] on linux via `pyenv`. I'm on WSLv1, with my `PATH` having the following:

```
/home/g-rath/.local/bin:/home/g-rath/.pyenv/plugins/pyenv-virtualenv/shims:/home/g-rath/.pyenv/shims:/home/g-rath/.pyenv/bin:/home/g-rath/.nodenv/shims:/home/g-rath/.nodenv/bin:/home/g-rath/.rbenv/shims:/home/g-rath/.rbenv/bin:/home/g-rath/.tfenv/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/c/Program Files/Docker/Docker/Resources/bin:/c/Program Files (x86)/Common Files/Oracle/Java/javapath:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0/:/c/Windows/System32/OpenSSH/:/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/c/Program Files/NVIDIA Corporation/NVIDIA NvDLISR:/c/bin:/c/Program Files/Git/cmd:/c/Program Files/Docker/Docker/resources/bin:/c/ProgramData/DockerDesktop/version-bin:/c/Program Files/dotnet/:/c/Program Files/PowerShell/7/:/c/Users/G-Rath/AppData/Local/Microsoft/WindowsApps:/c/Users/G-Rath/AppData/Local/hyper/app-3.0.2/resources/bin:/c/Program Files/JetBrains/IntelliJ IDEA 2019.3.1/bin:/snap/bin
```

`wslview` is opening Chrome, which I can't open directly from WSL (i.e `chrome.exe` results in "not found", so it isn't exposed on my PATH).

`wslview --version` gives `wslu v2.3.2; wslview v06`.

**Comment by mhils** [comment]
Thanks for the super quick turnaround times. I'm a bit at a loss here... do you have a BROWSER env var defined? What's your output of `which www-browser`? `xdg-settings get default-web-browser`?

**Comment by G-Rath** [comment]
> Thanks for the super quick turnaround times

Right back at you - I'm very keen to increase general WSL support where I can, and these kind of oddities are really useful to know about.

> do you have a BROWSER env var defined

I did, but I tested it both with and without that, and things worked.

However, I ran `pyenv rehash`, and now it's behaviouring the same as when you run the command, which is interesting 🤔 

```
g-rath@DESKTOP-MUGH739:/c/Users/G-Rath$ pyenv rehash
g-rath@DESKTOP-MUGH739:/c/Users/G-Rath$ python3 -c "import webbrowser; webbrowser.get('wslview')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/g-rath/.pyenv/versions/3.8.1/lib/python3.8/webbrowser.py", line 65, in get
    raise Error("could not locate runnable browser")
webbrowser.Error: could not locate runnable browser
```

**Comment by G-Rath** [comment]
Ok so it is the `BROWSER` env variable combined with running `pyenv rehash`.

**Comment by mhils** [comment]
Awesome, thanks! Let's update this PR to `wslview %s` and ship it? :)

**Comment by G-Rath** [comment]
Totally! sorry about that runaround - I'm still trying to think why that happened, but oh well 🤷 

I've tested that `wslview %s` works with the BROWSER env defined and not defined, so lets go for it!

**Comment by mhils** [comment]
Awesome, thanks! ❤️
